src/kontrol/kdist/assert.md

```diff
@@ -49,6 +50,14 @@ These macros expand into the `#assert` production defined earlier, providing an
     rule #assert_not_eq W1 W2 ERR => #assert (W1 =/=Int W2) ERR +Bytes String2Bytes(": " +String Int2String(W1) +String " == " +String Int2String(W2))
     rule #assert_approx_eq_abs W1 W2 W3 ERR => #assert ((maxInt(W1, W2) -Int minInt(W1, W2)) <=Int W3) ERR 
       +Bytes String2Bytes(": " +String Int2String(W1) +String " !~= " +String Int2String(W2) +String " (max delta: " +String Int2String(W3) +String ", real delta: " +String Int2String(maxInt(W1, W2) -Int minInt(W1, W2)) +String ")")
+    rule #assert_approx_eq_rel W1 W2 W3 ERR => #assert (W1 ==Int W2) ERR 
+      +Bytes String2Bytes(": " +String Int2String(W1) +String " !~= " +String Int2String(W2) 
+      +String " (max delta: " +String Int2String(W3 divInt (10 ^Int 16)) +String "% , real delta: undefined)") // displayed percentage in Kontrol does not include any decimal values
+      requires W2 ==Int 0
+    rule #assert_approx_eq_rel W1 W2 W3 ERR => #assert ((((maxInt(W1, W2) -Int minInt(W1, W2)) *Int (10 ^Int 18)) divInt absInt(W2)) <=Int W3) ERR 
```

Comment:
The implementation looks accurate to me, but I believe that it's just the multiplication and division that makes this particularly slow and difficult for the SMT solver. We've encountered that within the engagement, but I'm not sure if we've got any solutions for it now; let's check in on Monday.

I think, we're also doing this evaluation twice — once for the assert, and once for the resulting error message?

Reply:
Okay. Perhaps I will remove the resulting error message assertions from all the tests and put them in separate tests.